### ISSUE TITLE
Setting enableInteractiveSelection to false still allows cursor control via a11y on iOS

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#99284]
Internal: b/221156034

```dart
      return Scaffold(
        appBar: AppBar(),
        body: Column(children: [
          TextField(
            enabled: true,
            enableInteractiveSelection: false,
          ),
          TextField(
            enabled: true,
            enableInteractiveSelection: false,
          )
        ]),
      );
```

Steps to reproduce:

1. Enable accessibility mode
2. Focus on the second text field, and enter some text (abcd)
3. Swipe from bottom to top
    * Expected: cannot move the cursor
    * Actual: the accessibility focus is still on the second text field, and the cursor is moved (abc|d)

**Comment by goderbauer** [comment]
@chunhtai could you take a look at this customer issue?
